batch/sql/add-agg-billing-by-date.sql

```diff
@@ -0,0 +1,235 @@
+ALTER TABLE attempts ADD COLUMN dummy_aggregated_by_date INT DEFAULT 0, ALGORITHM=INSTANT;
+
+DROP TABLE IF EXISTS `attempts_aggregated_by_date`;
+CREATE TABLE IF NOT EXISTS `attempts_aggregated_by_date` (
+  `batch_id` BIGINT NOT NULL,
+  `job_id` INT NOT NULL,
+  `attempt_id` VARCHAR(40) NOT NULL,
+  PRIMARY KEY (`batch_id`, `job_id`, `attempt_id`),
+  FOREIGN KEY (`batch_id`, `job_id`, `attempt_id`) REFERENCES attempts(`batch_id`, `job_id`, `attempt_id`) ON DELETE CASCADE
+) ENGINE = InnoDB;
+
+DROP TABLE IF EXISTS `aggregated_billing_project_user_resources_by_date`;
+CREATE TABLE IF NOT EXISTS `aggregated_billing_project_user_resources_by_date` (
+  `billing_project` VARCHAR(100) NOT NULL,
+  `user` VARCHAR(100) NOT NULL,
+  `start_time` BIGINT NOT NULL,
+  `end_time` BIGINT NOT NULL,
+  `resource_id` INT NOT NULL,
+  `token` INT NOT NULL,
+  `usage` BIGINT NOT NULL DEFAULT 0,
+  PRIMARY KEY (`billing_project`, `user`, `start_time`, `end_time`, `resource_id`, `token`),
+  FOREIGN KEY (`billing_project`) REFERENCES billing_projects(name) ON DELETE CASCADE,
```

Comment:
Perhaps not on the critical path, but should we also make `billing_project` an integer ID here and not varchar?

Reply:
Hmmm.... my inclination is the storage savings is not going to be huge as this table is substantially smaller than the other aggregated_*_resources_by_date tables. But it would be good to avoid future headaches. Let's add it to this PR.

Reply:
Actually, we don't have an easy way to add an auto increment column for the `user` column as well since there's no users table. I think we should hold off on this change.

Reply:
Fine by me, we can keep it in mind for the future